numpy/lib/_iotools.py

```diff
@@ -838,13 +834,13 @@ def update(self, func, default=None, testing_value=None,
             self.type = self._dtypeortype(self._getdtype(default))
         else:
             try:
-                tester = func(testing_value or b'1')
+                tester = func(testing_value or '1')
             except (TypeError, ValueError):
                 tester = None
             self.type = self._dtypeortype(self._getdtype(tester))
         # Add the missing values to the existing set
         if missing_values is not None:
-            if _is_bytes_like(missing_values):
+            if isinstance(missing_values, basestring):
```

Comment:
This now rejects `bytes` on python 3. This is perhaps fine, but it does so silently, as there there is no `else` case to fail on this `if`

Reply:
Hmm, the whole function looks a bit bogus as it never checks what the iterator returns. `_is_bytes_like` also doesn't do much except check that it can be combined with the bytes type, which does not include `str` in Python 3.

The documentation does say that `missing_values` is a str or sequence of str, although it would originally accept unicode also. I think in Python 3 after this PR it should not accept bytes, but that might not be backwards compatible with third party converters as the original only worked with byte strings in Python 3.

Maybe the thing to do here is make `missing_values` iterable, if not already, check each value for basestring, and fail if not.